src/lib/radio/radio.ts

```diff
@@ -522,6 +548,15 @@ export class MatRadioButton extends _MatRadioButtonMixinBase
     this.change.emit(new MatRadioChange(this, this._value));
   }
 
+  private setColor(): void {
+    if (
+      !this.color &&
+      (this._providerOverride && this._providerOverride.color)
```

Comment:
Rather than having to check this every time, it might be easier to just set the initial value of `color` in the constructor to the one from the provider. The provider is unlikely to change after init.

Reply:
```
  ngDoCheck(): void {
    this.color = !this.color ?
    this._providerOverride && this._providerOverride.color
    : this.color;
  }
```
Currently this is what I use because when setting in constructor/ngOnInit - tests relating to `Expected every radio element to use the accent color by default` or `Expected every radio element to use the primary color from the binding` tend to fail.

Reply:
The best I can do is adding the provider check in `constructor` but the OnChanges still checks if color is set and applies, because it is possible to change the color `@Input()`.